src/Orleans.CodeGenerator/CopierGenerator.cs

```diff
@@ -53,8 +53,14 @@ internal static class CopierGenerator
                 Accessibility.Public => SyntaxKind.PublicKeyword,
                 _ => SyntaxKind.InternalKeyword,
             };
+
+            var isExceptionType = type.IsExceptionType && type.SerializationHooks.Count == 0;
```

Comment:
Is the complexity added by special casing worth it?

Reply:
This applies to all exception type copiers, which are generally not used, but still have a bunch of code generated for them (and it's usually identical). So it saves quite a lot of useless generated code.